teps/0082-workspace-hinting.md

```diff
@@ -0,0 +1,303 @@
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Workspaces allow Task authors to declare portions of their Task's filesystem to be supplied at runtime by TaskRuns or PipelineRuns. For example a Task may accept a credential via an optional Workspace and a TaskRun might supply it from a Secret. Another Task might write source code to a Workspace and a PipelineRun could bind a Persistent Volume to it so the source can be passed to other PipelineTasks.
+
+Rephrasing this slightly: the interface that Workspaces expose caters to a number of pretty disjoint use-cases - it's general-purpose. A down-side of that is Task authors can't communicate a Workspace's intended usage in a machine-readable way. There's no way for an author to indicate "this Workspace is intended to accept a credential" or "this Workspace should be supplied with configuration". Similarly for Pipeline authors, there's no way to hint that a Workspace is used to shuttle around between Tasks. They can write a human-readable description as part of the workspace declaration but that's essentially useless to an automated system constructing TaskRuns and PipelineRuns.
+
+The purpose of this TEP is to allow Task and Pipeline authors to "hint" about the intended purpose of a Workspace. The idea is that if authors can mark Workspaces with a purpose then automated systems could be designed to submit reasonable default bindings for them.
+
+## Motivation
+
+### Goals
+
+- Provide a way for Tasks and Pipelines to declare the purpose of Workspaces in a machine-readable format.
+
+### Non-Goals
+
+- Adding constraint-checking or any other logic to Pipelines to validate bound Workspaces based on workspace hints.
+
+### Use Cases (optional)
+
+The Tekton Workflows project is currently exploring ways to pass Secrets from a high-level Workflow description into a PipelineRun. This is made considerably more difficult because Pipelines can't indicate which of their Workspaces might be the right one to bind those Secrets to.
+
+## Requirements
+
+- Hinting must be optional: we don't want to suddenly invalidate every Task or Pipeline that currently includes a Workspace.
```

Comment:
kind of a tangent, but maybe this is something we'd eventually require in catalog workspaces 🤔  + @jerop